optimade/validator/validator.py

```diff
@@ -63,224 +60,97 @@
-                if not optional:
-                    validator.success_count += 1
-                else:
-                    validator.optional_success_count += 1
-                message = f"✔: {request} - {msg}"
-                if validator.verbosity > 0:
-                    if optional:
-                        print(message)
-                    else:
-                        print_success(message)
-                elif optional:
-                    print(".", end="", flush=True)
-                else:
-                    print_success(".", end="", flush=True)
-            else:
-                internal_error = False
-                request = request.replace("\n", "")
-                message = msg.split("\n")
-                if validator.verbosity > 1:
-                    # ValidationErrors from pydantic already include very detailed errors
-                    # that get duplicated in the traceback
-                    if not isinstance(result, ValidationError):
-                        message += traceback.split("\n")
-
-                if isinstance(result, InternalError):
-                    internal_error = True
-                    validator.internal_failure_count += 1
-                    summary = f"!: {request} - {test_fn.__name__} - failed with internal error"
-                    validator.internal_failure_messages.append((summary, message))
-                else:
-                    if not optional:
-                        validator.failure_count += 1
-                    else:
-                        validator.optional_failure_count += 1
-                    summary = f"✖: {request} - {test_fn.__name__} - failed with error"
-                    validator.failure_messages.append((summary, message))
-
-                if validator.verbosity > 0:
-                    if internal_error:
-                        print_notify(summary)
-                        for line in message:
-                            print_warning(f"\t{line}")
-                    elif optional:
-                        print(summary)
-                        for line in message:
-                            print(f"\t{line}")
-                    else:
-                        print_failure(summary)
-                        for line in message:
-                            print_warning(f"\t{line}")
-                else:
-                    if internal_error:
-                        print_notify("!", end="", flush=True)
-                    elif optional:
-                        print("✖", end="", flush=True)
-                    else:
-                        print_failure("✖", end="", flush=True)
+RESPONSE_CLASSES_INDEX = {
+    BASE_INFO_ENDPOINT: IndexInfoResponse,
+    LINKS_ENDPOINT: ValidatorLinksResponse,
+}
 
-                # set failure result to None as this is expected by other functions
-                result = None
+VERSIONS_REGEXP = r"/v[0-9]+(\.[0-9]+){,2}"
+
+PROPERTIES = {}
```

Comment:
This applies for all the following lines.

Reply:
I just built it this way without much thought, whilst hoping that we'd still be able to get stuff directly from the schema. It might be worth just dumping this whole thing to a JSON file to avoid cluttering everything up.

Reply:
I meant also mainly for defining `PROPERTIES`.
But while you mention it, yeah - many of these things should be moved to either `utils.py` or `config.py` or a `static.py` or something similar.